src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +392,420 @@ else if (!period.isEmpty()) {
         }
     }
 
+    /**
+     * Finds and return if summary tag present.
+     *
+     * @param javadoc Javadoc root node.
+     * @return true, if first sentence contains @summary tag.
+     */
+    private static boolean containsSummaryTag(DetailNode javadoc) {
+        boolean contains = false;
+        for (DetailNode node : javadoc.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                final DetailNode[] child = node.getChildren();
+                contains = isSummaryTag(child);
+            }
+        }
+
+        if (isSummaryTagWithinHtmlTag(javadoc)) {
```

Comment:
I do not understand why this method is required, you need single method that recurcively search for JAVADOC_INLINE_TAG in any tree.

Reply:
I tried implementing this using two ways
1. To reach every node recursively and check if it is a Javadoc node : The here is we are not guaranteed here that there is not string or HTML tags before inline tag. Also, it will lead to a poor memory management since we would be searching each and every javadoc node.
2. To use get the first child only and then similarly recursively go down the tree: This fails on many HTML tags will something have it own children as first child eg : `PARAGRAPH`

Since, these both failed I had to revert back to the previous method with better modification.